### PR TITLE
Adding the delete setter from kustomize

### PR DESCRIPTION
**Pull request by jijiew**
This pr:
1. add the delete setter from kustomize
2. add the related doc

### CODE REVIEW

**Comment by mortent** [comment]
This looks good to me. Seems like there is a merge conflict that needs to be resolved. 
Can we also squash the commits touching code, but keep the commit that updates the docs separate. And then add the `docs:` prefix to the commit message for docs so we can filter it out in the release notes.

File: internal/guides/generated/producer/substitutions.go [hunk 1]
```diff
@@ -156,7 +156,141 @@ substitution.
       spec:
         containers:
         - name: nginx
-          image: nginx:1.8.1 # {"$ref":"#/definitions/io.k8s.cli.substitutions.image-value"}
+          image: nginx:1.8.1 # {"$kpt-set":"image-value"}
+
+## Nested Substitutions
```
Comment (phanimarupaka) [inline internal/guides/generated/producer/substitutions.go:161]: Why is this part of this PR ? Isn't it already merged ?

**Comment by phanimarupaka** [comment]
Please rebase and explain why other merged docs are updated again ?

/lgtm

**Comment by jijiew** [comment]
> This looks good to me. Seems like there is a merge conflict that needs to be resolved.
> Can we also squash the commits touching code, but keep the commit that updates the docs separate. And then add the `docs:` prefix to the commit message for docs so we can filter it out in the release notes.

Done. thanks ;)

**Comment by jijiew** [comment]
> Please rebase and explain why other merged docs are updated again ?
> 
> /lgtm

I checked the master, it seams that it's not updated in the generated doc, it's possible that last person update the doc but forgot to run the command to update the generated doc, so when I run, it update that part as well.
https://github.com/GoogleContainerTools/kpt/blob/master/internal/guides/generated/consumer/substitute.go

File: internal/guides/generated/producer/substitutions.go (hunk 1 above)
Comment (jijiew) [reply internal/guides/generated/producer/substitutions.go:161]: I checked the master, it seams that it's not updated in the generated doc, it's possible that last person update the doc but forgot to run the command to update the generated doc, so when I run, it update that part as well.
https://github.com/GoogleContainerTools/kpt/blob/master/internal/guides/generated/consumer/substitute.go

**Comment by jijiew** [comment]
> This looks good to me. Seems like there is a merge conflict that needs to be resolved.
> Can we also squash the commits touching code, but keep the commit that updates the docs separate. And then add the `docs:` prefix to the commit message for docs so we can filter it out in the release notes.

Done. thanks ;)

File: go.mod [hunk 2]
```diff
@@ -25,6 +24,6 @@ require (
 	// Once there is a 0.18 release, we can import a semver release.
 	k8s.io/kubectl v0.0.0-20191219154910-1528d4eea6dd
 	sigs.k8s.io/cli-utils v0.15.0
-	sigs.k8s.io/kustomize/cmd/config v0.3.0
-	sigs.k8s.io/kustomize/kyaml v0.3.1-0.20200618190311-fb6830c98a78
+	sigs.k8s.io/kustomize/cmd/config v0.0.0-20200629173832-441581b74521
```
Comment (phanimarupaka) [inline go.mod:27]: What is this version ? Did this seep in by mistake ?

Comment (jijiew) [reply go.mod:27]: that's a new version I added by running go get sigs.k8s.io/kustomize/cmd/config of a specific sha with my delete setter change.

**Pull request merged**